src/lib/backend.ts

```diff
@@ -0,0 +1,457 @@
+    // more information see: https://goo.gl/nhqqwD
+    if (platform() === 'darwin' && process.type !== 'browser') {
+      return false;
+    }
+
+    return this.frontend.getOptions().enableNative !== false;
+  }
+
+  /** Activates the Electron CrashReporter. */
+  private async installNativeHandler(): Promise<boolean> {
+    // We are only called by the frontend if the SDK is enabled and a valid DSN
+    // has been configured. If no DSN is present, this indicates a programming
+    // error.
+    const dsn = this.frontend.getDSN();
+    if (!dsn) {
+      throw new SentryError(
+        'Invariant exception: install() must not be called when disabled',
+      );
+    }
+
+    // We will manually submit errors, but CrashReporter requires a submitURL in
+    // some versions. Also, provide a productName and companyName, which we will
+    // add manually to the event's context during submission.
+    crashReporter.start({
+      companyName: '',
+      ignoreSystemCrashHandler: true,
+      productName: getApp().getName(),
+      submitURL: '',
+      uploadToServer: false,
+    });
+
+    if (isMainProcess()) {
+      // The crashReporter has an undocumented method to retrieve the directory
+      // it uses to store minidumps in. The structure in this directory depends
+      // on the crash library being used (Crashpad or Breakpad).
+      const reporter: CrashReporterExt = crashReporter as any;
+      const crashesDirectory = reporter.getCrashesDirectory();
+
+      this.uploader = new MinidumpUploader(dsn, crashesDirectory, CACHE_PATH);
+
+      // Flush already cached minidumps from the queue.
+      forget(this.uploader.flushQueue());
+
+      // Start to submit recent minidump crashes. This will load breadcrumbs
+      // and context information that was cached on disk prior to the crash.
+      forget(this.sendNativeCrashes({ crashed_process: 'browser' }));
+
+      // Every time a subprocess or renderer crashes, start sending minidumps
+      // right away.
+      app.on('web-contents-created', (_, contents) => {
+        contents.on('crashed', async () => {
+          forget(
+            this.sendNativeCrashes({
+              crashed_process: `renderer[${contents.id}]`,
+              crashed_url: normalizeUrl(contents.getURL()),
+            }),
+          );
+        });
+      });
+    }
+
+    return true;
+  }
+
+  /** Activates the Node SDK for the main process. */
+  private async installMainHandler(): Promise<boolean> {
+    await this.frontend.setOptions({
+      onFatalError: async (error: Error) => {
+        console.error('*********************************');
+        console.error('* SentryElectron unhandledError *');
+        console.error('*********************************');
+        console.error(error);
+        console.error('---------------------------------');
+        await this.frontend.captureException(error);
+      },
+      ...this.frontend.getOptions(),
```

Comment:
I think we can remove this thanks to https://github.com/getsentry/raven-js/blob/4.0.0-wip/packages/core/src/lib/base.ts#L198